### ISSUE TITLE
The handling of features for cascading versions fails when a version is retired

### ISSUE THREAD

**Issue by afmenez** [compat-table/compat-table#1118]
The handling of features for cascading versions is failing when a version is retired.

For example, this is the current configuration for the IE10 browser on environments.json:
 ```
 "ie10": {
    "full": "Internet Explorer",
    "family": "Chakra",
    "short": "IE 10",
    "release": "2012-10-26",
    "retire": "2016-01-12",
    "obsolete": true,
    "test_suites": [
      "es5",
      "es6",
      "es2016plus",
      "esintl",
      "non-standard"
    ]
  },
```
Notice that it is present on the es2016plus tab, but not on the esnext tab.

Now this is a feature on the es2016plus tab:
```
  {
    name: 'assignments allowed in for-in head in non-strict mode',
    spec: 'https://tc39.github.io/ecma262/#sec-initializers-in-forin-statement-heads',
    mdn: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Compatibility_Initializer_expressions_in_strict_mode',
    category: '2017 annex b',
    significance: 'tiny',
    exec: function(){/*
     for (var i = 0 in {}) {}
     return i === 0;
     */},
    res: {
      tr: true,
      ie10: true,
      edge12: true,
      firefox2: true,
      firefox31: true,
      firefox40: false,
      firefox52: true,
      chrome30: true,
      safari3_1: true,
      safari9: false,
      safari10_1: true,
      safaritp: true,
      webkit: true,
      node0_12: true,
      android4_0: true,
      ios5_1: true,
      duktape2_0: true,
    },
  },
```
Notice that it is enabled for ie10 and not ie11, but the generator script takes care of this and cascades the feature for all versions >=10.  Which is great.

But what happens if you remove es2016plus from IE10 in the environments file?  Then IE10 is dropped from the es2016plus tab, as expected.  But also the feature above is marked as false for IE11, which is unexpected (and wrong!).

**Comment by afmenez** [comment]
The same problem happens when a browser is changed to obsolete: very.  It disappears from all tables, and its results are not cascaded to later versions.

**Comment by afmenez** [comment]
Well, I found that in both situations the browser version is dropped too early and its features are indeed not considered for cascading.  A better approach would be to add all browsers to the feature table, and when all the data is complete, remove obsolete and missing versions.

**Comment by ljharb** [comment]
That sounds like the right solution.

**Comment by afmenez** [comment]
~~The cascading is done in the function interpolateResults(), and the "very obsolete" browsers are dropped in the "var browser" definition just before it.  I can remove the filter to add all browsers and cascade the features correctly, but I have no idea how to remove them after interpolateResults() is done.~~
What I suggested above is already being made, there's two variables: browsers (filtered) and rawBrowsers (unfiltered).  So the problem must be somewhere else.
Regarding the other issue, "test_suites" filter is applied on the beginning of process.nextTick() function, but I don't know how to make it work.

**Comment by afmenez** [comment]
The comments above may imply that I am working on a solution, but I don't know enough to do it.  I'll leave this to someone more experienced.

**Comment by afmenez** [comment]
Fixed by #1600
